src/transformation/builtins/string.ts

```diff
@@ -46,35 +52,37 @@ export function transformStringPrototypeCall(
                 lua.SyntaxKind.SubtractionOperator,
                 node
             );
+        }
+
         case "substr":
-            if (node.arguments.length === 1) {
-                const argument = context.transformExpression(node.arguments[0]);
-                const arg1 = createExpressionPlusOne(argument);
-                return createStringCall("sub", node, caller, arg1);
-            } else {
-                const sumArg = lua.createBinaryExpression(params[0], params[1], lua.SyntaxKind.AdditionOperator);
-                return createStringCall("sub", node, caller, createExpressionPlusOne(params[0]), sumArg);
-            }
+            return transformLuaLibFunction(context, LuaLibFeature.StringSubstr, node, caller, ...params);
         case "substring":
-            if (node.arguments.length === 1) {
-                const arg1 = createExpressionPlusOne(params[0]);
-                return createStringCall("sub", node, caller, arg1);
-            } else {
-                const arg1 = createExpressionPlusOne(params[0]);
-                const arg2 = params[1];
-                return createStringCall("sub", node, caller, arg1, arg2);
-            }
-        case "slice":
-            if (node.arguments.length === 0) {
-                return caller;
-            } else if (node.arguments.length === 1) {
-                const arg1 = createExpressionPlusOne(params[0]);
-                return createStringCall("sub", node, caller, arg1);
-            } else {
-                const arg1 = createExpressionPlusOne(params[0]);
-                const arg2 = params[1];
-                return createStringCall("sub", node, caller, arg1, arg2);
+            return transformLuaLibFunction(context, LuaLibFeature.StringSubstring, node, caller, ...params);
+
+        case "slice": {
+            const literalArg1 = getNumberLiteralValue(params[0]);
+            if (params[0] && literalArg1 !== undefined) {
```

Comment:
Could you add a comment here explaining what this condition is and why it is added (also for `charAt` and `charCodeAt`). 
Something like `When slicing a string literal, translate to sliced Lua literal at compile-time to avoid lualib function call.`